spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/SparkTestHelperBase.java

```diff
@@ -69,19 +71,21 @@ protected void assertEquals(
   }
 
   protected void assertEquals(String context, Object[] expectedRow, Object[] actualRow) {
-    Assert.assertEquals("Number of columns should match", expectedRow.length, actualRow.length);
+    assertThat(actualRow).as("Number of columns should match").hasSameSizeAs(expectedRow);
     for (int col = 0; col < actualRow.length; col += 1) {
       Object expectedValue = expectedRow[col];
       Object actualValue = actualRow[col];
       if (expectedValue != null && expectedValue.getClass().isArray()) {
         String newContext = String.format("%s (nested col %d)", context, col + 1);
         if (expectedValue instanceof byte[]) {
-          Assert.assertArrayEquals(newContext, (byte[]) expectedValue, (byte[]) actualValue);
+          assertThat(actualValue).as(newContext).isEqualTo(expectedValue);
         } else {
           assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
         }
       } else if (expectedValue != ANY) {
-        Assert.assertEquals(context + " contents should match", expectedValue, actualValue);
+        assertThat(actualValue)
+            .as(context + " col " + (col + 1) + " contents should match")
```

Comment:
string concatenation is a bit difficult to read here, so maybe use string formatting instead? The `.as()` method supports string formatting

Reply:
Thanks for checking this and others. Let me update this part and the others.